code/modules/keybindings/bindings_client.dm

```diff
@@ -0,0 +1,112 @@
+// Clients aren't datums so we have to define these procs indpendently.
+// These verbs are called for all key press and release events
+/client/verb/keyDown(_key as text)
+	set instant = TRUE
+	set hidden = TRUE
+
+	client_keysend_amount += 1
+
+	var/cache = client_keysend_amount
+
+	if(keysend_tripped && next_keysend_trip_reset <= world.time)
+		keysend_tripped = FALSE
+
+	if(next_keysend_reset <= world.time)
+		client_keysend_amount = 0
+		next_keysend_reset = world.time + (1 SECONDS)
+
+	//The "tripped" system is to confirm that flooding is still happening after one spike
+	//not entirely sure how byond commands interact in relation to lag
+	//don't want to kick people if a lag spike results in a huge flood of commands being sent
+	if(cache >= MAX_KEYPRESS_AUTOKICK)
+		if(!keysend_tripped)
+			keysend_tripped = TRUE
+			next_keysend_trip_reset = world.time + (2 SECONDS)
+		else
+			log_admin("Client [ckey] was just autokicked for flooding keysends; likely abuse but potentially lagspike.")
+			message_admins("Client [ckey] was just autokicked for flooding keysends; likely abuse but potentially lagspike.")
+			qdel(src)
+			return
```

Comment:
We have systems for this; either replace or leave alone.

Reply:
Specifically `code/modules/admin/spam_prevention.dm`.